src/utils/actions.js

```diff
@@ -296,65 +298,119 @@ const actions = [
     translation: 'Object.ActionButton.ExecuteAutomation',
     value: 'executeAutomation',
     getActionCall:
-      ({ app, automation, automationPostData }) =>
-      async () => {
-        if (automation !== undefined) {
-          try {
-            automation = encodeURIComponent(automation);
-            const itemInfo = await fetch(`../api/v1/items/${automation}`).then((response) => response.json());
-            const autoInfo = await fetch(`../api/v1/automations/${itemInfo.resourceId}`).then((response) =>
-              response.json()
-            );
-            let executePath = `../api/v1/automations/${autoInfo.id}/actions/execute?X-Execution-Token=${autoInfo.executionToken}`;
+      ({ app, automation, automationId, automationTriggered, automationExecutionToken, automationPostData, buttonId, multiUserAutomation }) =>
+        async () => {
+          if (multiUserAutomation) {
+            let automationUrl
+            if (automation !== undefined) {
+              const a = await getAutomationFromItem(automation);
+              automationUrl = `../api/v1/automations/${a.id}/runs`;
+            }
+            else if (automationId !== undefined && automationTriggered) {
+              automationUrl = `../api/v1/automations/${automationId}/actions/execute?X-Execution-Token=${automationExecutionToken}`;
+            }
+            else if (automationId !== undefined && !automationTriggered) {
+              automationUrl = `../api/v1/automations/${automationId}/runs`;
+            }
+            else {
+              return;
+            }
+            let bookmark
             if (automationPostData) {
-              const inputBlocks = await fetch(`../api/v1/automations/${itemInfo.resourceId}/blocks`)
-                .then((response) => response.json())
-                .then((blocks) => {
-                  let items = [];
-                  for (let i = 0; i < blocks.blocks.length; i++) {
-                    if (blocks.blocks[i].type === 'FormBlock') {
-                      items = blocks.blocks[i].form;
-                      break;
-                    }
+              const bookmarkProps = {
+                qOptions: {
+                  qIncludeVariables: true,
+                  qIncludeAllPatches: true
+                }
+              };
+              bookmark = await app.createTemporaryBookmark(bookmarkProps);
+            }
+            const user = await getUser();
+            const inputs = {
+              app: app.id,
+              bookmark: automationPostData ? bookmark : '',
+              sheet: await getSheetId(app, buttonId),
+              user: user.subject,
+              space: await getSpaceId(app.id),
+              tenant: user.tenantId,
+              time: new Date(),
+            }
+            const automationData = {
+              guid: automationId,
```

Comment:
same comment here as above.

Reply:
I changed this to id both here and in the tests